app/template/page.tsx

```diff
@@ -0,0 +1,28 @@
+import { Metadata } from 'next';
+import PageClient, { IServerDataSchema } from './page.client';
+
+export const metadata: Metadata = {
+  title: 'EDPN Template',
+  description: 'Elite Dangerous Pilots Network',
+  icons: 'EDPN_logo_dark_background.png',
+};
+
+async function getData() {
+  // adjust revalidation time to your needs, or remove it completely
+  // here it's set to 1 day as the data doesn't change often
+  const req = await fetch(
+    `${process.env.NEXT_PUBLIC_STAGING_API_URL}/api/v1/trade/commodity/filter?type=WASTE&isRare=false`,
+    { next: { revalidate: 86400 } },
+  );
+
+  if (!req.ok) {
+    throw new Error('Failed to fetch data. Try again, or wait until later.');
+  }
```

Comment:
I understand what you are doing here and correct me if my assumption is wrong. Any failure here will show the closest error boundary. This is fine for now and I am not expecting any change in this PR, but ideally we want to create a typed schema for data that's sent to components as props. I haven't thought of the exact implementation but something on the lines of - if error on server side, set error and assign it to the component prop which can take success & error states so we can still show this alert of error in a localized section of the page. Let me think about it and create a discussion as an issue. Let me know your thoughts though.

Reply:
Correct about the error. This will go to the root error page you created recently.

To have what you're saying be useful, that is to pass the error along and provide it as feedback on the page, the error would have to be non-vital to the page's intent. There may be cases where that occurs, but atm I think all of our server fetches populate form fields that are required. Therefore, they are blockers to the form. But as the app grows, we may be happy to do what you are suggesting now, so I'm fine with that route too.

Reply:
It's good now for, we anyway need to think more about how we will handle that and if we even need it for a recoverable or non-critical error.